app/controllers/api/predictions_controller.rb

```diff
@@ -62,7 +64,12 @@ def find_prediction
     end
 
     def prediction_params
-      params.require(:prediction).permit!
+      permitted_params = params.require(:prediction).permit!
+      if permitted_params[:private].present?
```

Comment:
Oh I see you're handling that here. Might be worth a comment explaining what's happening here and why.